pkgs/test/nixpkgs-check-by-name/src/eval.rs

```diff
@@ -60,9 +58,9 @@ pub fn check_values(
         attrs_file_path.display()
     ))?;
 
+    let expr_path = std::env::var("NIX_CHECK_BY_NAME_EXPR_PATH")?;
```

Comment:
I was confused about this, thought we wanted the path to be linked at build time (compile time), although runtimepath does suggest otherwise lol, do we want runtime path to be linked so that development is easy? Isn't it better to simply nix run or something that takes care of it?

Reply:
Was confused about this, I thought we wanted to link it during compile time and keep everything in the store? We want to link it and still keep it outside store?

Im assuming it's to make development easy, but isn't it cleaner to do nix test or something for development?

Reply:
I assume you meant to reply to https://github.com/NixOS/nixpkgs/pull/267048#discussion_r1397974041

We only need it in the store so that the result can always find it without problems. It can't be a local (non-store) file there, since there's no guarantees about what the local filesystem looks like when the result is run.

But during development, we _do_ know the local filesystem, there's always going to be a `./src/eval.nix` file relative to the project root, so we can just use that instead. Does that make sense?

Reply:
Yea. Works.